### PR TITLE
[metrics] Force export census metrics on worker death

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

Census metrics are periodically pushed from individual workers to per-node metrics agent processes for aggregation. However, this can result in missing the latest metrics from workers (e.g., last few tasks that finished). Force an export during core worker shutdown to fix this.

This involves patch OCL to allow access to the ExportNow() API.

### CODE REVIEW

File: src/ray/common/ray_config_def.h [hunk 1]
```diff
@@ -460,7 +460,7 @@ RAY_CONFIG(int64_t, idle_worker_killing_time_threshold_ms, 1000)
 RAY_CONFIG(int64_t, num_workers_soft_limit, -1)
 
 // The interval where metrics are exported in milliseconds.
-RAY_CONFIG(uint64_t, metrics_report_interval_ms, 10000)
+RAY_CONFIG(uint64_t, metrics_report_interval_ms, 5000)
```
Comment (ericl) [inline src/ray/common/ray_config_def.h:463]: Unrelatedly, it seems 5 seconds is probably safe for now.

**Review by rkooo567 — APPROVED** [review]

File: python/ray/tests/test_task_metrics.py [hunk 2]
```diff
@@ -58,13 +59,45 @@ def f():
...
     }
-    # TODO(ekl) optimize the reporting interval to be faster for testing
     wait_for_condition(
         lambda: tasks_by_state(info) == expected, timeout=20, retry_interval_ms=500
     )
     proc.kill()
 
 
+def test_metrics_export_now(shutdown_only):
+    info = ray.init(num_cpus=2, **METRIC_CONFIG)
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/tests/test_task_metrics.py:69]: Will do

File: src/ray/core_worker/core_worker.cc [hunk 3]
```diff
@@ -590,6 +590,8 @@ void CoreWorker::Disconnect(
     const rpc::WorkerExitType &exit_type,
     const std::string &exit_detail,
     const std::shared_ptr<LocalMemoryBuffer> &creation_task_exception_pb_bytes) {
+  // Force stats export before exiting the worker.
+  opencensus::stats::StatsExporter::ExportNow();
```
[the remark this answers is not in the document]
Comment (ericl) [reply src/ray/core_worker/core_worker.cc:594]: Yes, I checked here and OCL acquires a mutex internally.

Regarding the RPC safety, I think at least the RPC initiation is a blocking call. I'm not sure if we wait for a reply though.

[the remark this answers is not in the document]
Comment (rkooo567) [reply src/ray/core_worker/core_worker.cc:594]: Ah, I see. If the initiation is a blocking call, I think it is pretty safe although we are not waiting for a reply...

**Pull request merged**